### PR TITLE
Add therading signal

### PR DESCRIPTION
**Pull request by quatchi2010**

### CODE REVIEW

File: src/twinkie_console.c [hunk 1]
```diff
@@ -74,31 +87,31 @@ void *control_d(void *p) {
...
-    if (file_is_open) {
-      time(&timer);
-      tm = localtime(&timer);
-      /* alternatively read file name from something */
-      sprintf(file_name, "./%s/%d_%d_%d_%d_%d_%d.bin", dir_name,
-              tm->tm_year + 1900, tm->tm_mon, tm->tm_mday, tm->tm_hour,
-              tm->tm_min, tm->tm_sec);
-      file_open(file_name);
-    } else {
-      file_close();
```
Comment (ualbertagreen) [inline src/twinkie_console.c:89]: Oh I just noticed I don't send the shell commands correctly when opening and closing the files. I've pushed a patch to correctly turn the twinkie on and off. If you rebase off of my patch it should be more clear where to go from there.

Sorry about the confusion.

File: src/twinkie_console.c [hunk 2]
```diff
@@ -74,31 +87,31 @@ void *control_d(void *p) {
...
+  time(&timer);
+  tm = localtime(&timer);
+  /* alternatively read file name from something */
+  sprintf(file_name, "./%s/%d_%d_%d_%d_%d_%d.bin", dir_name, tm->tm_year + 1900,
+          tm->tm_mon, tm->tm_mday, tm->tm_hour, tm->tm_min, tm->tm_sec);
+  file_open(file_name);
+  printf("\nFile Opened\n");
+
+  pthread_cond_wait(&cond1, &lock);
+  printf("sned stop");
```
Comment (ualbertagreen) [inline src/twinkie_console.c]: send

**Review by ualbertagreen — COMMENTED** [review]
I don't recommend completely opening and closing the app for every log. This is because the twinkie needs a bit of time to clean up everytime the data stream connects or disconnects. If connection is interrupted and re-established repeatedly, there might be a problem where the app attempts a connection before the twinkie is ready and doesn't find the twinkie.

Best practice is to open the connection when the twinkie is plugged in to the host machine and maintain the connection until all tests are completed.

simply use the `start` and `stop` shell commands in between tests.

also because the twinkie is constantly caching old pd packets in the background even if it should be stopped, the `reset` shell command can also be used between tests to completely wipe the twinkie memory.

File: src/twinkie_console.c (hunk 2 above)
Comment (quatchi2010) [reply src/twinkie_console.c]: Fixed.

File: src/twinkie_console.c (hunk 1 above)
Comment (quatchi2010) [reply src/twinkie_console.c:89]: NP

**Pull request merged**